x-pack/test_serverless/api_integration/test_suites/common/platform_security/roles_routes_feature_flag.ts

```diff
@@ -5,58 +5,1003 @@
+                },
+              ],
+              applications: [
+                {
+                  application: 'kibana-.kibana',
+                  privileges: ['read'],
+                  resources: ['*'],
+                },
+                {
+                  application: 'apm',
+                  privileges: ['apm-privilege'],
+                  resources: ['*'],
+                },
+              ],
+              metadata: {
+                bar: 'old-metadata',
+              },
+            },
+          });
+
+          await supertestWithoutAuth
+            .put('/api/security/role/role_to_update')
+            .set('kbn-xsrf', 'xxx')
+            .set(svlCommonApi.getInternalRequestHeader())
+            .set(roleAuthc.apiKeyHeader)
+            .send({
+              metadata: {
+                foo: 'test-metadata',
+              },
+              elasticsearch: {
+                cluster: ['manage'],
+                indices: [
+                  {
+                    names: ['logstash-*'],
+                    privileges: ['read', 'view_index_metadata'],
+                  },
+                ],
+                remote_indices: [
+                  {
+                    clusters: ['remote_cluster1'],
+                    names: ['remote_index1', 'remote_index2'],
+                    privileges: ['all'],
+                  },
+                ],
+              },
+              kibana: [
+                {
+                  feature: {
+                    dashboard: ['read'],
+                    dev_tools: ['all'],
+                  },
+                  spaces: ['*'],
+                },
+                {
+                  base: ['all'],
+                  spaces: ['marketing', 'sales'],
+                },
+              ],
+            })
+            .expect(400);
+
+          const role = await es.security.getRole({ name: 'role_to_update' });
+          expect(role).to.eql({
+            role_to_update: {
+              cluster: ['monitor'],
+              indices: [
+                {
+                  names: ['beats-*'],
+                  privileges: ['write'],
+                  allow_restricted_indices: false,
+                },
+              ],
+              applications: [
+                {
+                  application: 'kibana-.kibana',
+                  privileges: ['read'],
+                  resources: ['*'],
+                },
+                {
+                  application: 'apm',
+                  privileges: ['apm-privilege'],
+                  resources: ['*'],
+                },
+              ],
+              metadata: {
+                bar: 'old-metadata',
+              },
+              run_as: [],
+              transient_metadata: {
+                enabled: true,
+              },
+            },
+          });
+        });
+      });
+
+      describe('Delete Role', () => {
+        it('should delete the roles we created', async () => {
```

Comment:
**note:** I have mixed feelings about this one. Ideally, the test shouldn't depend on other tests to pass because if the tests that this one depends on fail, it becomes hard to figure out what's broken - whether it's the delete functionality or create/update. I'd prefer to delete roles in `afterEach` and re-create them within this test, but I won't block it if you feel it's acceptable.

Reply:
I agree with you! I 'borrowed' this implementation from the stateful test suite, see: https://github.com/elastic/kibana/blob/c07e946e0cc66db5ca9a6b7256f0f6b73ece04ea/x-pack/test/api_integration/apis/security/roles.ts#L479

That said, we shouldn't perpetuate this methodology, so I will update the serverless tests accordingly.